### ISSUE TITLE
Add option to set content of parent_page from master_doc

### ISSUE THREAD

**Issue by hason** [sphinx-contrib/confluencebuilder#439]
Assume that Sphinx generates follow structure:
```
master_doc
|- page 1
|  |- subpage 1
|- page 2
```

If I set `confluence_parent_page` configuration that confluence structure is as same as:
```
parent_page
|- master_doc
|  |- page 1
|  |  |- subpage 1
|  |- page 2
```

I would like to have a configuration to unite *parent_page* with *master_doc*:

```
parent_page (content from master_doc)
|- page 1
|  |- subpage 1
|- page 2
```

**Comment by jdknight** [comment]
This should be achievable. A thing to consider is what to do about applying the main document's title value when publishing. The value of the configuration `confluence_parent_page` is the name of the document to publish to, but the publication of the main document is determined from the detected title value inside the main document. It is assumed with this feature that the idea is that the main document's title will not be considered when publishing. For example, if the main documents title is "test-a", the target page to publish to is "test-b" and this proposed feature is enabled, the final page root will still be "test-b".

Some things to consider:
- Ensuring we track that name of the target document's name to ensure links to the main document are still properly referenced.
- We could permit the naming of the target page and instead rework `confluence_parent_page_id_check` to be something like `confluence_parent_page_id`. A publish configuration only needs to care about the identifier of the page and whatever name from the documentation set can still be used. Although, maybe it would be better if we type-checked the `confluence_parent_page` value -- where a string would indicate a page name and an integer would indicate a page identifier.

**Comment by jdknight** [comment]
Changes introduced in #463 should provide the ability for a user to explicitly set the root page to update, instead of using a container page (`confluence_parent_page`) to use as a parent. Users should be able to explicitly set the page ID to update their root document on as follows (replacing the example `12345` value with the page identifier reported by the Confluence instance being published to):

```
confluence_publish_root = 12345
```

This feature is available on the main branch and should be made available next stable release (v1.5).

Note that users who wish to emulate this capability without explicitly configuring `confluence_publish_root` or users on v1.4 and older can achieve this by:
1. Making a publish event without setting a `confluence_parent_page`/`confluence_publish_root` value.
2. Moving the published root into the desired location inside a Confluence space.

Any future publish events will update the existing published pages and not change the ancestry of the main document.

**Comment by jdknight** [comment]
v1.5 is now available on PyPI -- marking as closed.
